### DISCUSSION TITLE
mainImage for using depth does not work

### DISCUSSION THREAD

**Discussion by aliiscripts** [pmndrs/postprocessing#300] [Q&A]
the regular `void mainImage(const in vec4 inputColor, const in vec2 uv, out vec4 outputColor);` works fine but I want to use the one that includes depth ( `void mainImage(const in vec4 inputColor, const in vec2 uv,
	const in float depth, out vec4 outputColor);` ) as a parameter but get this error: _'e1MainImage' : no matching overloaded function found_ . I assume it because that function doesn't exist but not sure how to fix, any help would be appreciated.

**Comment by vanruesc** [comment]
Make sure your effect is using `EffectAttribute.DETPH`:

```js
import { Effect, EffectAttribute } from "postprocessing";

class MyEffect extends Effect {

	constructor() {

		super("MyEffect", fragmentShader, {
			attributes: EffectAttribute.DEPTH
		});

	}

}
```
